### ISSUE TITLE
Exiting with exception: NotFoundError(/home/oms/projects/montblanc/montblanc/ext/rime.so: undefined symbol: _ZN10tensorflow15OpKernelContext21CtxFailureWithWarningENS_6StatusE)

### ISSUE THREAD

**Issue by o-smirnov** [ska-sa/montblanc#262]
Can't get current master to work on 18.04. Pip install not an option due to #261. As a kludge, I cloned master, removed the astropy dependency from setup.py (system has astropy installed anyway), and went the ``setup.py develop --user`` route. This did install tensorflow etc. But now I get:

```
 - 16:45:48 - main               [2.8/3.0 6.6/6.6 3.0Gb] Exiting with exception: NotFoundError(/home/oms/projects/montblanc/montblanc/ext/rime.so: undefined symbol: _ZN10tensorflow15OpKernelContext21CtxFailureWithWarningENS_6StatusE)
 Traceback (most recent call last):
  File "/home/oms/projects/CubiCal/cubical/main.py", line 360, in main
    stats_dict = workers.run_process_loop(ms, tile_list, load_model, single_chunk, solver_type, solver_opts, debug_opts)
  File "/home/oms/projects/CubiCal/cubical/workers.py", line 207, in run_process_loop
    return _run_single_process_loop(ms, load_model, single_chunk, solver_type, solver_opts, debug_opts)
  File "/home/oms/projects/CubiCal/cubical/workers.py", line 316, in _run_single_process_loop
    tile.load(load_model=load_model)
  File "/home/oms/projects/CubiCal/cubical/data_handler/ms_tile.py", line 814, in load
    model = subset.load_montblanc_models(uvwco, loaded_models, model_source, cluster, imod, idir)
  File "/home/oms/projects/CubiCal/cubical/data_handler/ms_tile.py", line 228, in load_montblanc_models
    MBTiggerSim.simulate(srcs, snks, self.tile.dh.mb_opts)
  File "/home/oms/projects/CubiCal/cubical/data_handler/MBTiggerSim.py", line 250, in simulate
    _mb_slvr = montblanc.rime_solver(slvr_cfg)
  File "/home/oms/projects/montblanc/montblanc/__init__.py", line 107, in rime_solver
    return montblanc.factory.rime_solver(slvr_cfg)
  File "/home/oms/projects/montblanc/montblanc/factory.py", line 23, in rime_solver
    from montblanc.impl.rime.tensorflow.RimeSolver import RimeSolver
  File "/home/oms/projects/montblanc/montblanc/impl/rime/tensorflow/RimeSolver.py", line 53, in <module>
    rime = load_tf_lib()
  File "/home/oms/projects/montblanc/montblanc/impl/rime/tensorflow/__init__.py", line 30, in load_tf_lib
    return tf.load_op_library(rime_lib_path)
  File "/home/oms/.local/lib/python2.7/site-packages/tensorflow/python/framework/load_library.py", line 60, in load_op_library
    lib_handle = py_tf.TF_LoadLibrary(library_filename)
NotFoundError: /home/oms/projects/montblanc/montblanc/ext/rime.so: undefined symbol: _ZN10tensorflow15OpKernelContext21CtxFailureWithWarningENS_6StatusE
```

**Comment by o-smirnov** [comment]
``pip install -e montblanc`` inside a virtualenv produces the same error. This is with tensorflow-1.8.0.

**Comment by sjperkins** [comment]
Might be a C++ ABI issue with tensorflow on 18.04. If this isn't a
completely new virtual environment I would recommend starting a new one
from scratch.

**Comment by sjperkins** [comment]
> Tried that in a brand-new virtual env. Same problem.

Try the following:

1. Clone montblanc master
2. Modify this [line](https://github.com/ska-sa/montblanc/blob/72a8cc48f94fc6af22a500dfa25cdc42114bd551/install/tensorflow_ops_ext.py#L111) to `('_GLIBCXX_USE_CXX11_ABI', 1)]`
3. pip install -e <clone-location>

**Comment by sjperkins** [comment]
I've installed master on both ray (18.04) and stevie (16.04.5). Both installs succeed and example scripts run.

**Comment by o-smirnov** [comment]
The fix I pushed to #261 also resolves this. No idea why, but I'll take it!

**Comment by sjperkins** [comment]
Likely caused by the build process not realising the old object + shared object files need to be replaced when a new tensorflow version is installed. I'll try and think of a reasonable way of supporting this.
